types/object-hash/index.d.ts

```diff
@@ -7,7 +7,7 @@
 // Minimum TypeScript Version: 3.9
 
 declare namespace objectHash {
-    type NotUndefined = Record<string, unknown> | string | number | boolean | null | NotUndefined[];
+    type NotUndefined = object | string | number | boolean | null | NotUndefined[];
```

Comment:
I would suggest to use 
```suggestion
    type NotUndefined = { [x: string]: unknown } | string | number | boolean | null | NotUndefined[];
```
Instead of 'object'

Reply:
@martin-badin Your suggested change would have the same problem as `Record<string, unknown>`.  Please see this [TS Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZgVzMAqmA1mEB3MMC8MAFCAEYBWAXDAEoCmoATgCYA80jAlmAOYA0MJJhxgAfAEoComAG8AUDEUxQkEABs6AOjUgeJCuIDccgL5y5K6PCTAAkmGZ0AHgGVOPAsTJVZMANpO1BzcPAC61EJYuDAmkvjS8krK4BDqWjp63kam5pawiMgAgmAAnp76PvRMbMG8AgCGpRJSsgpKlmnauhXZZhYp+TYA8hQMsIQV1N5jcQltih0aXZkGxn3cUHSMcPXAdDAAYiAgrUmk9YxBUFy8xmf1AF7UYAgAtqRba-2Q+cfURydCIklOdLjAAOSg8F8eYwc5PGAARgAzMiAOw5AqoDBRMBEODHIwwAD0xJgW0YIEYcix9kcrnc+MJhhJZIpVJpNmKJSZICJpJg2Cp6AgnOQI3IY15-LJQsYIvMUBKAAd9gAhC6eYGKUFXG48O4gx7PN4fRhfPJwi7UDWMLWw3UQqEw+4IlHonJi7HCXBEUEywXC0W0hzONx6f0sgVyhVY7l+i4BmPB4ajYBQBOMJNBoA) for a few examples of different functions accepting interfaces and types.  As you'll see, only `Record<string, any>` and `object` work for both `type` and `interface`.

I had actually created a [discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/63625#discussioncomment-4412710) about the interface issue before I realized that @JarnoRFB had already created this PR.  I've tried to illustrate there as best as I can why `object` is likely the most viable option here for the type definition.

Thanks!